apps/dashboard/src/components/workflow-editor/steps/configure-inline/configure-step-inline-form.tsx

```diff
@@ -0,0 +1,94 @@
+import { useCallback, useEffect, useMemo } from 'react';
+import merge from 'lodash.merge';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import type { StepDataDto, StepIssuesDto, UpdateWorkflowDto, WorkflowResponseDto } from '@novu/shared';
+import { StepTypeEnum } from '@novu/shared';
+
+import { flattenIssues, updateStepControlValuesInWorkflow } from '@/components/workflow-editor/step-utils';
+import { buildDefaultValues, buildDefaultValuesOfDataSchema, buildDynamicZodSchema } from '@/utils/schema';
+import { Form } from '@/components/primitives/form/form';
+import { useFormAutosave } from '@/hooks/use-form-autosave';
+import { SaveFormContext } from '@/components/workflow-editor/steps/save-form-context';
+import { DelayForm } from '@/components/workflow-editor/steps/delay/delay-form';
+
+const STEP_TYPE_TO_INLINE_FORM: Record<StepTypeEnum, (args: StepInlineFormProps) => React.JSX.Element | null> = {
```

Comment:
This code along with `calculateDefaultValues` is duplicated between this new component and `configure-step-template-form.tsx` We need to make it dry.

Reply:
I agree.
Also, it will result in the form nested under another form, which we wanted to avoid.
The `configure-step-form` should just register all the fields under the `controls` in the form I guess, because this is the shape of the `step` and that will be updated later with using the `update workflow` endpoint.
```
{
   "name": "In-App Step",
   "stepId": "some-step-id",
   "controls": { ... } 
}

Reply:
Later here we will also need to connect `custom step controls` from the code created workflows.